base/show.jl

```diff
@@ -1350,19 +1374,40 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
         show_list(io, args, ", ", indent)
 
     elseif head === :macrocall && nargs >= 2
-        # first show the line number argument as a comment
-        if isa(args[2], LineNumberNode) || is_expr(args[2], :line)
-            print(io, args[2], ' ')
-        end
-        # Use the functional syntax unless specifically designated with prec=-1
-        # and hide the line number argument from the argument list
-        if prec >= 0
-            show_call(io, :call, args[1], args[3:end], indent)
+        # handle some special syntax
+        # x"y" and x"y"z
+        if occursin(r"^Symbol\(\"@[a-zA-Z].*_str\"\)$",
+                    repr(args[1]))
```

Comment:
I think `endswith(string(args[1]::Symbol), "_str")` should be sufficient; would be better not to depend on regexes too much in this code.

Reply:
Ok, modified accordingly.